storage/innobase/row/row0mysql.cc

```diff
@@ -1781,6 +1790,50 @@ init_fts_doc_id_for_ref(
 	}
 }
 
+/** System Versioning: modify update vector to set row_start
+ * (or row_end in case of DELETE) to current trx_id. */
+void upd_node_t::vers_set_fields(const trx_t* trx)
+{
+	dict_index_t* clust_index = dict_table_get_first_index(table);
+	upd_field_t* ufield;
+	dict_col_t* col;
+	unsigned col_idx;
+	if (is_delete) {
+		ufield = &update->fields[0];
+		update->n_fields = 0;
+		is_delete = VERSIONED_DELETE;
+		col_idx = table->vers_end;
+	} else {
+		ut_ad(update->n_fields < table->n_cols);
+		ufield = &update->fields[update->n_fields];
+		col_idx = table->vers_start;
+	}
+	col = &table->cols[col_idx];
+	UNIV_MEM_INVALID(ufield, sizeof *ufield);
+	{
+		ulint field_no = dict_col_get_clust_pos(col, clust_index);
+		ut_ad(field_no != ULINT_UNDEFINED);
+		ufield->field_no = field_no;
+	}
+	ufield->orig_len = 0;
+	ufield->exp = NULL;
+
+	if (col->vers_native())
+	{
+		mach_write_to_8(update->vers_sys_value, trx->id);
+	} else {
+		thd_get_query_start_data(trx->mysql_thd, (char *)
+					update->vers_sys_value);
```

Comment:
Do not use C-style casts.
Use `reinterpret_cast<char*>`. Why can’t `vers_sys_value` not be declared in the matching type in the first place?